cms/apps/pages/models.py

```diff
@@ -176,6 +155,40 @@ def navigation(self):
         help_text="Visitors that aren't logged in won't see this page in the navigation"
     )
 
+    class Meta:
+        unique_together = (('parent', 'slug', 'country_group'),)
+        ordering = ('left',)
+
+    class MPTTMeta:
+        right_attr = 'right'
+        left_attr = 'left'
+
+    def save(self, *args, **kwargs):
+        super(Page, self).save(*args, **kwargs)
+        Page.objects.rebuild()
+
+    def get_absolute_url(self):
+        if not self.parent:
+            return urls.get_script_prefix()
+
+        return self.parent.get_absolute_url() + self.slug + '/'
+
+    @cached_property
+    def children(self):
+        '''The child pages for this page.'''
+        children = []
+        if self.right - self.left > 1:  # Optimization - don't fetch children
+            #  we know aren't there!
+            for child in self.child_set.filter(is_content_object=False):
+                child.parent = self
+                children.append(child)
+        return children
```

Comment:
Would the `get_children()` method of MPTTModel not work here?

Reply:
It would. I might need to include the filter on it though